app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/signatures/GetSignaturesDialog.kt

```diff
@@ -0,0 +1,186 @@
+import com.babylon.wallet.android.presentation.accessfactorsources.composables.RoundLedgerItem
+import com.babylon.wallet.android.presentation.ui.composables.BottomSheetDialogWrapper
+import com.babylon.wallet.android.utils.BiometricAuthenticationResult
+import com.babylon.wallet.android.utils.biometricAuthenticate
+import com.babylon.wallet.android.utils.formattedSpans
+import com.radixdlt.sargon.FactorSource
+import com.radixdlt.sargon.annotation.UsesSampleValues
+import rdx.works.core.sargon.sample
+
+@Composable
+fun GetSignaturesDialog(
+    modifier: Modifier = Modifier,
+    viewModel: GetSignaturesViewModel,
+    onDismiss: () -> Unit
+) {
+    val state by viewModel.state.collectAsState()
+    val context = LocalContext.current
+
+    BackHandler {
+        viewModel.onUserDismiss()
+    }
+
+    LaunchedEffect(Unit) {
+        viewModel.oneOffEvent.collect { event ->
+            when (event) {
+                is GetSignaturesViewModel.Event.RequestBiometricToAccessDeviceFactorSource -> {
+                    context.biometricAuthenticate { biometricAuthenticationResult ->
+                        when (biometricAuthenticationResult) {
+                            BiometricAuthenticationResult.Succeeded -> viewModel.collectSignaturesForDeviceFactorSource(
+                                deviceFactorSource = event.deviceFactorSource,
+                                signers = event.signers,
+                                signRequest = event.signRequest
+                            )
+                            BiometricAuthenticationResult.Error -> { /* do nothing */ }
+                            BiometricAuthenticationResult.Failed -> { /* do nothing */ }
+                        }
+                    }
+                }
+
+                GetSignaturesViewModel.Event.AccessingFactorSourceCompleted -> onDismiss()
+                GetSignaturesViewModel.Event.UserDismissed -> onDismiss()
+            }
+        }
+    }
+
+    GetSignaturesBottomSheetContent(
+        modifier = modifier,
+        showContentForFactorSource = state.showContentForFactorSource,
+        onDismiss = viewModel::onUserDismiss,
+        onRetryClick = viewModel::onRetryClick
+    )
+}
+
+@Composable
+private fun GetSignaturesBottomSheetContent(
+    modifier: Modifier = Modifier,
+    showContentForFactorSource: GetSignaturesViewModel.State.ShowContentForFactorSource,
+    onDismiss: () -> Unit,
+    onRetryClick: () -> Unit
+) {
+    BottomSheetDialogWrapper(
+        modifier = modifier,
+        onDismiss = {
+            onDismiss()
+        }
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = RadixTheme.dimensions.paddingXLarge)
+                .background(RadixTheme.colors.defaultBackground),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Spacer(Modifier.height(40.dp))
```

Comment:
If you rebased this branch with my changes on main, I think we should get rid of those fixed height spacers and add   parameter `heightFraction = 0.7f` to `BottomSheetDialogWrapper`, so we have consistent sheet size and spacing

Reply:
done ✅